inception-imls-stringmatch/src/main/java/de/tudarmstadt/ukp/inception/recommendation/imls/stringmatch/StringMatchingRecommender.java

```diff
@@ -84,6 +107,53 @@ public StringMatchingRecommender(Recommender aRecommender,
         traits = aTraits;
         gazeteerService = aGazeteerService;
     }
+    
+    @Override
+    public void export(RecommenderContext aContext, OutputStream aTarget) throws IOException
+    {
+
+        aContext.get(KEY_MODEL);
+        aContext.get(KEY_TAGSET);
+
+        Trie<DictEntry> dict = aContext.get(KEY_MODEL).orElseGet(this::createTrie);
+
+        User user = userService.getCurrentUser();
+
+        CAS cas = new CASImpl();
+
+        List<CAS> casList=new ArrayList<>();
+        casList.add(cas);
+
+        try {
+            train(aContext, casList );
+        }
+        catch (RecommendationException e) {
+            e.printStackTrace();
+        }
+
+        Predictions predictions = recommendationService.getPredictions(user, new Project());
+
+        // TODO #176 use the document Id once it it available in the CAS
+        String sourceDocumentName = CasMetadataUtils.getSourceDocumentName(cas)
+                .orElse(getDocumentTitle(cas));
+
+        // Extract all predictions for the current document / recommender
+        List<AnnotationSuggestion> suggestions = predictions.getPredictions().entrySet().stream()
+                .filter(f -> f.getKey().getDocumentName().equals(sourceDocumentName))
+                .filter(f -> f.getKey().getRecommenderId() == this.recommender.getId().longValue())
+                .map(Map.Entry::getValue)
+                .filter(s -> s.isVisible())
+                .collect(Collectors.toList());
+        
+        try {
+        	ObjectOutputStream objos= new ObjectOutputStream(new FileOutputStream("D://recommender.txt"));
```

Comment:
Hardcoded file and path is not good... I think the intended behaviour is that the user should be prompted by the browser to store or view the exported model file.

Reply:
We usually use an `AjaxDownloadLink` when a user should export something:

https://github.com/inception-project/inception/blob/b578054f77891bb152c22bb3a9ee8e3126d31a7c/inception-ui-kb/src/main/java/de/tudarmstadt/ukp/inception/ui/kb/project/AccessSpecificSettingsPanel.java#L342-L343